tests/Unit/TestSuite.php

```diff
@@ -11,3 +12,11 @@
     $this->expectExceptionMessage(sprintf('A test with the description `%s` already exist in the filename `%s`.', 'foo', __FILE__));
     $testSuite->tests->set(new \Pest\Factories\TestCaseFactory(__FILE__, 'foo', $test));
 });
+
+it('alerts users about tests with arguments but no input', function () {
+    $testSuite = new TestSuite(getcwd(), 'tests');
+    $test = function (int $arg) {};
+    $this->expectException(DatasetMissing::class);
+    $this->expectExceptionMessage(sprintf("A test with the description '%s' has %d argument(s) ([%s]) and no dataset(s) provided in %s", 'foo', 1, 'int $arg', __FILE__));
+    $testSuite->tests->set(new \Pest\Factories\TestCaseFactory(__FILE__, 'foo', $test));
+});
```

Comment:
Is it worth using the `->throws()` syntax here, or will that not work for some reason? 🤔

```suggestion
    $testSuite->tests->set(new \Pest\Factories\TestCaseFactory(__FILE__, 'foo', $test));
})->throws(
    DatasetMissing::class,
    sprintf("A test with the description '%s' has %d argument(s) ([%s]) and no dataset(s) provided in %s", 'foo', 1, 'int $arg', __FILE__)
);
```

Reply:
Good idea. I copied the above test down. I gave it the same treatment – seems to work just fine. 👍